ui/src/app/applications/components/applications-list/applications-filter.tsx

```diff
@@ -105,16 +105,38 @@ export class ApplicationsFilter extends React.Component<ApplicationsFilterProps,
                     values.add(app.metadata.labels[label]);
                 })
             );
+
+        const filtersCount = AppsListPreferences.countEnabledFilters(pref);
+
         return (
             <div className={classNames('applications-list__filters-container', {'applications-list__filters-container--expanded': this.state.expanded})}>
-                <i
-                    onClick={() => this.setState({expanded: !this.state.expanded})}
-                    className={classNames('fa applications-list__filters-expander', {'fa-chevron-up': this.state.expanded, 'fa-chevron-down': !this.state.expanded})}
-                />
-                <p className='applications-list__filters-container-title'>Filter By:</p>
-                <div className='row'>
+                <div className={classNames('applications-list__filters-title', {'applications-list__filters-title--expanded': this.state.expanded})}>
+                    <Tooltip content='Filters'>
+                        <i className='fa fa-filter' />
+                    </Tooltip>
+                    <div style={{marginLeft: 'auto'}} />
+                    <div className='applications-list__filters-title__counter tags-input__tag' style={filtersCount === 0 ? {display: 'none'} : {}}>
+                        {filtersCount} Filter{filtersCount === 1 ? '' : 's'}
+                        <Tooltip content='Clear Filters'>
+                            <i
+                                className='fa fa-times'
+                                onClick={() => {
+                                    AppsListPreferences.clearFilters(this.props.pref);
```

Comment:
I've reproduced a bug. 

1. Navigate to the applications list 
2. Add project filter
3. Click "Clear filters"

Projects tags input control keep showing the selected project. I think this is an old bug in https://github.com/argoproj/argo-cd/blob/master/ui/src/app/shared/components/tags-input/tags-input.tsx

`getDerivedStateFromProps()` handler is missing and component does not support changing input

Reply:
I believe I've fixed it - I should have noticed that one when testing.